config/management/operational/src/validator_set.rs

```diff
@@ -18,9 +24,56 @@ pub struct ValidatorSet {
 }
 
 impl ValidatorSet {
-    pub fn execute(self) -> Result<Vec<ValidatorInfo>, Error> {
+    pub fn execute(self) -> Result<Vec<DecryptedValidatorInfo>, Error> {
         let config = self.config.load()?.override_json_server(&self.json_server);
         let client = JsonRpcClientWrapper::new(config.json_server);
-        client.validator_set(self.account_address)
+        client.validator_set(self.account_address).map(|vec| {
+            vec.iter()
+                .map(|info| {
+                    DecryptedValidatorInfo::from_validator_info(
+                        &TEST_SHARED_VAL_NETADDR_KEY,
+                        0,
+                        info,
+                    )
+                    .unwrap()
+                })
+                .collect()
+        })
+    }
+}
+
+#[derive(Serialize)]
+pub struct DecryptedValidatorInfo {
+    pub account_address: AccountAddress,
+    // Validator config
```

Comment:
might be more effective if you had ValidatorConfig but a single noun like this isn't particularly clear what it implies... from my high context perspective I take it to mean the following fields are derived from the validator config;

Reply:
Yeah, they're derived from ValidatorConfig, and one point it was the validator config, but it's been cut out.